### PR TITLE
Make swift techspec compliant

### PR DESCRIPTION
**Pull request by epwn**
Change the permissions of swift logs and swift config files to be in
compliance with the techspec document

### CODE REVIEW

File: roles/swift-common/tasks/main.yml [hunk 1]
```diff
@@ -76,6 +76,13 @@
     - logging
   when: logging.enabled|default('True')|bool
 
+- name: change swift log file permissions
+  file: path={{ item }} mode=0644 owner=swift group=swift state=touch
+  with_items:
+    - /var/log/swift/account.log
+    - /var/log/swift/container.log
+    - /var/log/swift/object.log  
+
```
Comment (ryshah) [inline roles/swift-common/tasks/main.yml]: Need to add /var/log/swift/proxy.log

Comment (ryshah) [inline roles/swift-common/tasks/main.yml]: Actually you should just use variable for log paths defined in  https://github.com/blueboxgroup/ursula/blob/e3d19e68f8f0b42b05d0a4b34dc7eec7755ac8d4/roles/swift-common/defaults/main.yml

Comment (j2sol) [inline roles/swift-common/tasks/main.yml]: That's a good note, and probably should be done in future techspec PRs (and retroactively in previous changes).

Comment (paulczar) [inline roles/swift-common/tasks/main.yml]: currently these logs should be created with 0640 owned by syslog:adm ( inherited from the directory perms https://github.com/blueboxgroup/ursula/blob/master/roles/swift-common/tasks/logging.yml#L2-L3 )

does settings it to 0644 by swift:swift improve security ?   How will syslog write to them without write permissions ?

File: roles/swift-common/tasks/main.yml [hunk 2]
```diff
@@ -76,6 +76,11 @@
     - logging
   when: logging.enabled|default('True')|bool
 
+- name: change swift log file permissions
+  file: path={{ item }} mode=0644 owner=swift group=swift state=touch
+  with_items:
+    - swift-logs-paths
```
Comment (paulczar) [inline roles/swift-common/tasks/main.yml]: where is list coming from ?

File: roles/swift-common/tasks/main.yml [hunk 3]
```diff
@@ -76,6 +76,11 @@
     - logging
   when: logging.enabled|default('True')|bool
 
+- name: change swift log file permissions
+  file: path={{ item }} mode=0644 owner=swift group=swift state=touch
```
Comment (paulczar) [inline roles/swift-common/tasks/main.yml]: the swift logging directory is being created [here](https://github.com/blueboxgroup/ursula/blob/master/roles/swift-common/tasks/logging.yml#L2-L3) owned by syslog:adm which allows syslog to write to the logfiles ( which would inherit the perms from the dir to be 640 [i think] to syslog:adm as they get created by syslog ).

The perms listed here would mean that syslog would not be able to log on behalf of swift which would likely then break swift.

Comment (epwn) [inline roles/swift-common/tasks/main.yml]: we should discuss a better approach on slack Tuesday. I agree, we wouldnt want to break syslog.

File: roles/swift-common/tasks/main.yml (hunk 2 above)
Comment (epwn) [inline roles/swift-common/tasks/main.yml]: https://github.com/blueboxgroup/ursula/blob/e3d19e68f8f0b42b05d0a4b34dc7eec7755ac8d4/roles/swift-common/defaults/main.yml

File: roles/swift-common/tasks/main.yml [hunk 4]
```diff
@@ -76,6 +76,16 @@
...
 
+- name: Change cinder swift dir acl
+  acl: name=/var/log/swift state=present default=yes etype={{ item.etype }} permissions={{ item.permission }}
+  with_items:
+    - etype: user
+      permission: rw
+    - etype: group
+      permission: r
+    - etype: other
+      permission: r
```
Comment (j2sol) [inline roles/swift-common/tasks/main.yml]: This task should be done in the logging task file, so that it can be picked up with the `logging` tag.

**Comment by j2sol** [comment]
Just relocate the task and this will be good to test.

File: roles/swift-common/tasks/main.yml (hunk 4 above)
Comment (epwn) [inline roles/swift-common/tasks/main.yml]: also that weird 'cinder swift' typo in the name will be removed

**Comment by j2sol** [comment]
ok to test

**Pull request merged**